impedance/models/circuits/circuits.py

```diff
@@ -404,11 +403,13 @@ def __init__(self, CPE=False, **kwargs):
 
         circuit_len = calculateCircuitLength(self.circuit)
 
-        assert len(self.initial_guess) + len(self.constants) == circuit_len, \
-            'The number of initial guesses + ' + \
-            'the number of constants ({})'.format(len(self.initial_guess)) + \
-            ' needs to be equal to ' + \
-            'the circuit length ({})'.format(circuit_len)
+        if len(self.initial_guess) + len(self.constants) != circuit_len:
+            raise ValueError('The number of initial guesses ' +
+                             f'({len(self.initial_guess)}) + ' +
+                             'the number of constants ' +
+                             f'({len(self.constants)})' +
+                             ' must be equal to ' +
+                             f'the circuit length ({circuit_len})')
 
 
 class CustomCircuit(BaseCircuit):
```

Comment:
Sort of tangential, but can we make `initial_guess` an explicit keyword argument? If they're omitted an error is raised, so making it explicit could communicate the requirement better to the user.

Reply:
Challenge here is in loading a circuit from a .json file. It we want .load to be a method of BaseCircuit, then we need to be able to create a circuit instance (without an initial guess) and then call .load(filename). The alternative is a separate function (which is how it was until recently), but I thought it was neater to have it live with the circuit model class so I moved it in the refactor. 

To me, it makes sense to be able to create a circuit without an initial guess, but we have to make sure that we raise a meaningful error if the user tries to fit or predict before supplying one.

Reply:
Got it. I'm Ok living with it as-is, then.